src/services/formatting/ruleOperationContext.ts

```diff
@@ -4,20 +4,27 @@
 namespace ts.formatting {
 
     export class RuleOperationContext {
-        private customContextChecks: { (context: FormattingContext): boolean; }[];
-
-        constructor(...funcs: { (context: FormattingContext): boolean; }[]) {
-            this.customContextChecks = funcs;
+        constructor(private optionName: string, private checkApplyRuleOperation: { (optionName: string, options: FormatCodeSettings): boolean }, private customContextChecks: { (context: FormattingContext): boolean; }[]) {
         }
 
-        static Any: RuleOperationContext = new RuleOperationContext();
+        static Any: RuleOperationContext = new RuleOperationContext(undefined, undefined, []);
 
+        static Create1(...funcs: { (context: FormattingContext): boolean; }[]) {
+            return new RuleOperationContext(undefined, undefined, funcs);
+        }
+        static Create2(optionName: string, checkApplyRuleOperation: { (optionName: string, options: FormatCodeSettings): boolean }, ...funcs: { (context: FormattingContext): boolean; }[]) {

```

Comment:
consider not adding the new two options here, and leaving it as a set of functions as it was.

Under this assumption, your input for:
```ts
Create2("insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces", Rules.IsOptionEnabledOrUndefined, Rules.IsBraceWrappedContext)
```

would be:
```ts
new RuleOperationContext(Rules.IsOptionEnabledOrUndefined("insertSpaceAfterOpeningAndBeforeClosingNonemptyBraces"), Rules.IsBraceWrappedContext)
```

where `IsOptionEnabledOrUndefined` is defined as:

```ts
    static IsOptionEnabledOrUndefined(optionName: string): (option) => boolean {
             return (options) => !options.hasOwnProperty(optionName) || (<any>options)[optionName];
    }
```